src/MeetingsSDKAdapter.js

```diff
@@ -1031,31 +1028,40 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    *
    * @param {string} ID Meeting ID
    * @param {string} cameraID ID of the camera to switch to
-   * @private
    */
   async switchCamera(ID, cameraID) {
-    const sdkMeeting = this.fetchMeeting(ID);
-    const {stream, permission} = await this.getStream(
-      ID,
-      {sendVideo: true},
-      {video: {deviceId: cameraID}},
-    ).toPromise();
+    await this.updateMeeting(ID, async (meeting) => {
+      let updates;
+      const {stream, permission} = await this.getStream(
+        ID,
+        {sendVideo: true},
+        {video: {deviceId: cameraID}},
+      ).toPromise();
+
+      if (stream) {
+        updates = {
+          localVideo: {
+            stream,
+            permission: meeting.localVideo.permission,
```

Comment:
deepMerge is needed to avoid listing unrelated properties here. What if another prop is added to localVideo? one would have to come here and add it to this list of "updates", which is prone to errors.